.gitignore

```diff
@@ -0,0 +1,31 @@
+# Numerous always-ignore extensions
+*.diff
+*.err
+*.orig
+*.log
+*.rej
+*.swo
+*.swp
+*.vi
+*~
+
+# OS or Editor files
+.DS_Store
+.cache
+.project
+.settings
+.tmproj
+nbproject
+Thumbs.db
+.idea/
+
+# NPM packages folder
+node_modules/
+
+# Code coverage report
+.nyc_output/
+coverage/
+coverage.lcov
+
+# Test file - who doesn't have a test file?
+test.js
```

Comment:
Hahaha! Let me explain: I usually start a new project by first - I know, rather lazily - copying all the boilerplate from one of my other recent projects. I never spend much attention to the `.gitignore`, nor have I ever had this initial step peer-reviewed. This file is the result of years of copying from one project to the next, with slight modifications in many steps along the way. I think it started as a file generated by https://www.gitignore.io/, which explains all the editor related stuff.

You make very valid points though, and I will clean it up promptly! :flushed: